artemis-server/src/main/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImpl.java

```diff
@@ -306,6 +306,8 @@
 
    private Long globalMaxSize;
 
+   private boolean amqpTreatRejectAsUnmodifiedDeliveryFailed = ActiveMQDefaultConfiguration.getDefaultTreatRejectAsUnmodifiedDeliveryFailed();
```

Comment:
I would use the ProtocolManager configuration, being a property of the acceptor, instead of a global property for such thing.

That way you can configure this as part of the acceptor properties.

Reply:
Are you changing this similar to your other featrure? add it to ProtonProtocolManager instead?

Reply:
Done